src/wp-includes/class-wp-block.php

```diff
@@ -607,3 +571,121 @@ public function render( $options = array() ) {
 		return $block_content;
 	}
 }
+
+// phpcs:disable Generic.Files.OneObjectStructurePerFile.MultipleFound
+/**
+ * HTML API Processor subclass implementing experimental set_inner_html.
+ *
+ * DO NOT USE THIS CLASS. Internal usage only.
+ *
+ * @access private
+ */
+class PrivateProcessor extends WP_HTML_Processor {
+
+	public function set_inner_html( ?string $html ) {
```

Comment:
This method allows does "naive" inner HTML replacement, where the content that should be inside the node can change the HTML tree beyond the node.

This is also true with the existing implementation so it's not a regression and can be dealt with now or in follow-up work.